modules/dists/BlockDist.chpl

```diff
@@ -1742,14 +1750,11 @@ proc BlockArr.doiScan(op, dom) where (rank == 1) &&
 
   // Store one element per locale in order to track our local total
   // for a cross-locale scan as well as flags to negotiate reading and
-  // writing it.  This domain really wants an easier way to express
-  // it...
-  use ReplicatedDist;
+  // writing it.
   const ref targetLocs = this.dsiTargetLocales();
-  const elemPerLocDom = {1..1} dmapped Replicated(targetLocs);
-  var elemPerLoc: [elemPerLocDom] resType;
-  var inputReady$: [elemPerLocDom] sync bool;
-  var outputReady$: [elemPerLocDom] sync bool;
+  var elemPerLoc: [targetLocs.domain] resType;
+  var inputReady$: [targetLocs.domain] sync bool;
+  var outputReady$: [targetLocs.domain] unmanaged BoxedSync(bool)?;
```

Comment:
It's been argued (and I believe it) that we should/could get out of the habit of using `$` as a signifier for sync variables since there are no implicit reads/writes to them anymore.  You don't need to make the change here, but if you're in agreement with that and want to start the reversal...

Reply:
Dropped `$` in https://github.com/chapel-lang/chapel/pull/20185